packages/itwin/grouping-mapping-widget/src/widget/components/GroupPropertyAction.tsx

```diff
@@ -204,55 +310,135 @@ const extractProperties = (
               nestedContentField.pathToPrimaryClass[0].relationshipInfo.name ===
               "BisCore:GeometricElement3dHasTypeDefinition"
             ) {
-              const className =
-                nestedContentField.pathToPrimaryClass[0].targetClassInfo.name;
-              extractProperties(
-                nestedContentField.nestedFields,
-                classToPropertiesMapping,
-                {
-                  navigationName: "TypeDefinition",
-                  rootClassName: className,
-                }
-              );
-              // Hardcoded BisCore navigation properties for the type definition.
-              classToPropertiesMapping.get(className)?.push({
-                name: "TypeDefinition.Model.ModeledElement.UserLabel",
-                label: "Model UserLabel (TypeDefinition)",
-                type: "string",
-              });
-
-              classToPropertiesMapping.get(className)?.push({
-                name: "TypeDefinition.Model.ModeledElement.CodeValue",
-                label: "Model CodeValue (TypeDefinition)",
-                type: "string",
-              });
-
+              ecPropertyMetaDetaList.push(...extractNested(["TypeDefinition"], nestedContentField.nestedFields));
             }
             break;
           }
           default: {
             // Some elements don't have a path to primary class or relationship meaning..
             // Most likely a simple struct property
             if (!nestedContentField.pathToPrimaryClass) {
-              const columnName = (property as PropertiesField).properties[0]
-                .property.name;
-              const className = (property as PropertiesField).properties[0]
-                .property.classInfo.name;
-              extractStructProperties(
-                navigation
-                  ? `${navigation.navigationName}.${columnName}`
-                  : columnName,
-                navigation ? navigation.rootClassName : className,
-                classToPropertiesMapping,
-                property.type.members
-              );
-
+              extractStruct(property, ecPropertyMetaDetaList);
             }
           }
         }
       }
     }
   }
+  return ecPropertyMetaDetaList;
+};
+
+const convertToECProperties = (property: PropertyMetaData): Array<ECProperty> => {
+  const ecProperty: ECProperty = {
+    ecSchemaName: property.schema,
+    ecClassName: property.className,
+    ecPropertyType: property.type,
+    ecPropertyName: "",
+  };
+  switch (property.primitiveNavigationClass) {
+    // Hardcode Models navigation path and label behavior
+    case "BisCore:ModelContainsElements": {
+      return [
+        {
+          ...ecProperty,
+          ecPropertyName: [
```

Comment:
Keep the formatting consistent. Here you add each array element to a new line. Below you keep everything on a single line. Pick one.

Reply:
Done